connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -616,20 +627,24 @@ function removeAllButLastOccurences(deltaList: microsoftgraph.DriveItem[]) {
  * The function makes the assumption that there is no circular parent
  * relationship
  */
-function sortForIncrementalUpdate(changedList: DriveItem[], rootId: string) {
+function sortForIncrementalUpdate(changedList: DriveItem[], rootId?: string) {
   if (changedList.length === 0) {
     return [];
   }
 
   const internalIds = changedList.map((item) => getDriveItemInternalId(item));
 
   const sortedItemList = changedList.filter((item) => {
-    if (item.id === rootId) {
+    if (!item.parentReference) {
+      return false;
+    }
+
+    if (rootId && item.id === rootId) {
       // Found selected root
       return true;
     }
 
-    if (!item.parentReference || !item.parentReference.id) {
+    if (!rootId && !item.parentReference.id) {
```

Comment:
what about using `item.root` for testing if it is the root drive VS existence of a parentref id? it's the graph api field expliciltly for that

Reply:
also, update the comment below (since we don't always include the root)